src/Compilers/Core/MSBuildTask/Microsoft.CSharp.Core.targets

```diff
@@ -3,10 +3,18 @@
 <Project ToolsVersion="14.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <Import Project="Microsoft.Managed.Core.targets"/>
 
-  <PropertyGroup Condition="('$(TargetFrameworkIdentifier)' != '.NETCoreApp' OR '$(_TargetFrameworkVersionWithoutV)' &lt; '3.0') AND
-                            ('$(TargetFrameworkIdentifier)' != '.NETStandard' OR '$(_TargetFrameworkVersionWithoutV)' &lt; '2.1')">
-    <MaxSupportedLangVersion Condition="'$(MaxSupportedLangVersion)' == ''">7.3</MaxSupportedLangVersion>
-    <LangVersion Condition="'$(LangVersion)' == ''">$(MaxSupportedLangVersion)</LangVersion>
+  <PropertyGroup>
+    <!-- .NETCoreApp < 3.0, .NETStandard < 2.1, or any other target framework -->
+    <MaxSupportedLangVersion Condition="('$(TargetFrameworkIdentifier)' != '.NETCoreApp' OR '$(_TargetFrameworkVersionWithoutV)' &lt; '3.0') AND
+                                        ('$(TargetFrameworkIdentifier)' != '.NETStandard' OR '$(_TargetFrameworkVersionWithoutV)' &lt; '2.1') AND
+                                        '$(MaxSupportedLangVersion)' == ''">7.3</MaxSupportedLangVersion>
+    
+    <!-- .NETCoreApp < 5.0, .NETStandard == 2.1 -->
+    <MaxSupportedLangVersion Condition="('$(TargetFrameworkIdentifier)' != '.NETCoreApp' OR '$(_TargetFrameworkVersionWithoutV)' &lt; '5.0') AND
+                                        ('$(TargetFrameworkIdentifier)' != '.NETStandard' OR '$(_TargetFrameworkVersionWithoutV)' == '2.1') AND
+                                        '$(MaxSupportedLangVersion)' == ''">8.0</MaxSupportedLangVersion>
+    
```

Comment:
We should add a section that makes 9.0 the default when targeting netcoreapp 5.0

Reply:
Hmm, I was originally thinking we should just make the default language version be 9.0 given that we're now explicitly setting to 8 or less for anything other than .netcoreapp 5.0. 

I guess we can temporarily default netcoreapp 5.0 to csharp 9, and then let it go back to default when we release?

Reply:
The way I think about it, at the MSBuild level, is that the default is 7.3. Yet there are specific TF well known TF where we will change the default to a different value. So `net5` will always  have the value 9, it will never have the default.

Reply:
Hmm, I guess I was thinking there can be multiple netcoreapp5 versions, like 5.1/5.2 etc that should also be C# 9, but given that we're switching to a yearly 5/6/7 cadence, I guess that'll no longer be true? At which point there is a single one-to-one mapping of supported tfm <=> langversion right?